src/xtgeo/grid3d/_egrid.py

```diff
@@ -120,7 +124,7 @@ class Filehead:
     grid_format: GridFormat
 
     @classmethod
-    def from_egrid(cls, values: list[int]):
+    def from_egrid(cls, values: list[int]) -> Filehead:
```

Comment:
Probably want to use `typing_extensions.Self` here.

Reply:
Thats cool, was not aware of this feature.